Fix chunk size for newer decks

jianli opened the pull request:
When viewing deck statistics with "deck life" selected, the value of the X-axis intervals was hard-coded to 31 (a month). For a month-old deck, for example, this results in a chart with only a single bar. This diff keeps the chunk size to 1 (a day) for decks less than 100 days old, and to 7 (a week) for decks less than 100 weeks old. It also standardizes inconsistent values used for the end date (30 or 31 days) when "1 month" is selected.

dae commented:
Thanks, would you mind sending me a message about the license as mentioned in the LICENSE file?

I think the chunk size should be larger however - having many little chunks can lead to bad performance on slower devices, so something like 7 days for 1mo-1yr and 31 days after that might be better.

dae commented:
The build tests also appear to be failing, which may have been something you missed or a result of the recent fixes I've applied - could you please take a look?

jianli commented:
Hi Damien,

Thanks for your work in creating and maintaining Anki!

> Thanks, would you mind sending me a message about the license as mentioned in the LICENSE file?

Sent.

> I think the chunk size should be larger however - having many little chunks can lead to bad performance on slower devices, so something like 7 days for 1mo-1yr and 31 days after that might be better.

Since I've found it useful to see daily stats beyond one month, how about making the transitions so that there are at most 100 bins? So daily up to 100 days, then weekly up to 100 weeks, then monthly afterwards.

>The build tests also appear to be failing, which may have been something you missed or a result of the recent fixes I've applied - could you please take a look?

I think I broke them initially. Fixed and rebased.

jianli commented on anki/stats.py:
```diff
@@ -269,34 +283,25 @@ def _due(self, start=None, end=None, chunk=1):
-            yaxes=[dict(min=0), dict(position="right",min=0)])
+            yaxes=[dict(min=0), dict(position="right", min=0)])
         if days is not None:
             conf['xaxis']['min'] = -days+0.5
```
The linter is complaining about this line:

```
anki/stats.py:294:35: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
```

but this seems both spurious and unrelated to this pull request.

dae commented:
Thanks, 100 sounds like a good compromise, and I've fixed the linting issue - I'll get this merged in the next few days.
